lib/index.js

```diff
@@ -20,42 +20,42 @@ exports.register.attributes = {
     pkg: require('../package.json')
 };
 
+internals.schema = Joi.object({
+    cookie: Joi.string().default('sid'),
+    password: Joi.string().required(),
+    ttl: Joi.number().integer().min(0).when('keepAlive', { is: true, then: Joi.required() }),
+    domain: Joi.string(),
+    path: Joi.string().default('/'),
+    clearInvalid: Joi.boolean().default(false),
+    keepAlive: Joi.boolean().default(false),
+    isSecure: Joi.boolean().default(true),
+    isHttpOnly: Joi.boolean().default(true),
+    redirectTo: Joi.string(),
+    appendNext: Joi.boolean().default(false),
+    redirectOnTry: Joi.boolean().default(true),
+    validateFunc: Joi.func()
+}).required();
 
 internals.implementation = function (server, options) {
 
-    Hoek.assert(options, 'Missing cookie auth strategy options');
-    Hoek.assert(!options.validateFunc || typeof options.validateFunc === 'function', 'Invalid validateFunc method in configuration');
-    Hoek.assert(options.password, 'Missing required password in configuration');
-    Hoek.assert(!options.keepAlive || options.ttl, 'Cannot configure keepAlive without ttl');
+    var results = Joi.validate(options, internals.schema);
+    Hoek.assert(!results.error, results.error);
 
-    var settings = Hoek.clone(options);                        // Options can be reused
-    settings.cookie = settings.cookie || 'sid';
+    var settings = results.value;
 
     var cookieOptions = {
         encoding: 'iron',
         password: settings.password,
-        isSecure: settings.isSecure !== false,                  // Defaults to true
-        path: '/',
-        isHttpOnly: settings.isHttpOnly !== false,              // Defaults to true
+        isSecure: settings.isSecure,                  // Defaults to true
+        path: settings.path,
+        isHttpOnly: settings.isHttpOnly,              // Defaults to true
         clearInvalid: settings.clearInvalid,
-        ignoreErrors: true
+        ignoreErrors: true,
+        domain: settings.domain,
```

Comment:
It is optional, when settings.domain will be set to undefined if the user did not provide a domain. This undefined value will be passed on to server.state() and it will behave as if it were not provided at all. I dug through the server.state() code and found the Joi schema :

hapi/node_modules/statehood/lib/index.js
```javascript
    path: Joi.string().allow(null),
    domain: Joi.string().allow(null),
```
If you really want to, I can specify null to be the default value for ttl and domain, but it doesn't really change anything functionally as far as I can tell. Please advise.